### PR TITLE
Fix error when output is set to name

### LINKED ISSUE #754: kn service list -o name doesn't seem to work
### Bug report

```
$ kn service list -o name
list types are not supported by name printing: *v1.ServiceList
```

### Expected behavior

Show me the list of services by name - e.g.: `service.serving.knative.dev/echo`

### Steps to reproduce the problem

```
kn service create echo --image duglin/echo
kn service list -o name
kn service list echo -o name
```

Both of the last 2 cmds should work

### kn version

master/HEAD

### Knative (serving/eventing) version

master/HEAD

### PR DESCRIPTION
**Pull request by daisy-ycguo**
## Describe

Fix the error when calling `kn service list -o name`. Because the default name printer only supports `unstructured.UnstructuredList`, we have to convert `servingv1.ServiceList` to `unstructured.UnstructuredList` when output flag is set.

## Changes

* Convert servingv1.ServiceList to unstructured.UnstructuredList when output flag is set in `kn service list -o name`

## Reference

Fixes #754

### CODE REVIEW

File: pkg/util/unstructured.go [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+		}
+
+	} else {
+
+	}
+	return unstructuredList, nil
+
+}
+
+func toUnstructured(obj runtime.Object) (*unstructured.Unstructured, error) {
```
Comment (navidshaikh) [inline pkg/util/unstructured.go:48]: see if you can reuse https://github.com/knative/pkg/blob/master/apis/duck/unstructured.go#L32

Comment (daisy-ycguo) [reply pkg/util/unstructured.go:48]: Thanks for the information.
I checked pkg/apis/duck, and noticed it only accepts the type of `[duck.OneOfOurs](https://github.com/knative/pkg/blob/master/apis/duck/interface.go#L34)`. Unfortunately, neither v1.ServiceList nor Service even runtime.Object cannot satisfy that interface.

File: pkg/util/unstructured.go [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+	"encoding/json"
+
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// ToUnstructuredList is to converts an object to unstructured.UnstructuredList.
+// If the object is not a list type, it will convert to a single item UnstructuredList.
+func ToUnstructuredList(obj runtime.Object) (*unstructured.UnstructuredList, error) {
```
[the remark this answers is not in the document]
Comment (daisy-ycguo) [reply pkg/util/unstructured.go:27]: see pkg/util/unstructured_test.go

File: pkg/kn/commands/service/list.go [hunk 3]
```diff
@@ -81,11 +82,15 @@ func NewServiceListCommand(p *commands.KnParams) *cobra.Command {
 				return a.ObjectMeta.Name < b.ObjectMeta.Name
 			})
 
-			err = printer.PrintObj(serviceList, cmd.OutOrStdout())
-			if err != nil {
-				return err
+			if serviceListFlags.GenericPrintFlags.OutputFlagSpecified() {
+				unstructedList, err := util.ToUnstructuredList(serviceList)
```
[the remark this answers is not in the document]
Comment (daisy-ycguo) [reply pkg/kn/commands/service/list.go:86]: resolved.

**Comment by daisy-ycguo** [comment]
@navidshaikh This PR is ready for review.

**Pull request merged**